sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -280,6 +280,29 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   }
 
   /**
+   * Extracts the [[StructField]] with the given name recursively.
+   *
+   * @throws IllegalArgumentException if the parent field's type is not StructType
+   */
+  def getFieldRecursively(name: String): StructField = {
```

Comment:
The mark of nested fields is one kind of tmp data, finally it will convert to a pruned StructType and pass to `org.apache.spark.sql.parquet.row.requested_schema`